src/util/error.cpp

```diff
@@ -67,32 +105,28 @@ bool TError::Deserialize(int fd, TError &error) {
     if (ret == 0)
         return false;
     if (ret != sizeof(Error)) {
-        error = TError(EError::Unknown, errno, "Can't deserialize error");
+        error = FromErrno("Can't deserialize error");
         return true;
     }
     ret = read(fd, &errno, sizeof(Errno));
     if (ret != sizeof(Errno)) {
-        error = TError(EError::Unknown, errno, "Can't deserialize errno");
+        error = FromErrno("Can't deserialize errno");
         return true;
     }
     ret = read(fd, &len, sizeof(len));
     if (ret != sizeof(len)) {
-        error = TError(EError::Unknown, errno, "Can't deserialize length");
+        error = FromErrno("Can't deserialize length");
         return true;
     }
 
-    char *buf = new char[len];
-    ret = read(fd, buf, len);
+    std::string desc(len, '\0');
+    ret = read(fd, &desc[0], len);
     if (ret != len) {
-        delete[] buf;
-        error = TError(EError::Unknown, errno, "Can't deserialize description");
+        error = FromErrno("Can't deserialize description");
         return true;
     }
 
-    desc = std::string(buf, len);
-    delete[] buf;
-
-    error = TError(err, desc, errno);
+    error = TError(err, std::move(desc), errno);
```

Comment:
This function looks strange - it returns true even if it fails to read a TError value from file and assign to result value errno code.
If reading error appears, it will be impossible to if it is a deserialized error or it is a generated error for read operation.